### PR TITLE
Permission checks

### PR DESCRIPTION
**Pull request by NoComment1105**
Adds a check to member checks for `Permissions` set so users don't have to spam `hasPermission()` in their check blocks when checking for multiple permissions

### CODE REVIEW

File: kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/checks/MemberChecks.kt [hunk 1]
```diff
@@ -113,3 +115,100 @@ public suspend fun CheckContext<*>.notHasPermission(perm: Permission) {
         }
     }
 }
+
+/**
+ * Check asserting that the user an [Event] fired for has a given permission set, or the Administrator permission.
```
Comment (gdude2002) [inline kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/checks/MemberChecks.kt]: Very small thing - "the given permissions", not "a given permission".

File: kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/checks/MemberChecks.kt [hunk 2]
```diff
@@ -113,3 +115,100 @@ public suspend fun CheckContext<*>.notHasPermission(perm: Permission) {
...
+                    "checks.hasPermissions.failed",
+                    replacements = arrayOf(perms.values.forEach { it.translate(locale) })
+                )
+            )
+        }
+    }
+}
+
+/**
+ * Check asserting that the user an [Event] fired for **does not have** a given permission set **or** the Administrator
```
Comment (gdude2002) [inline kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/checks/MemberChecks.kt]: Very small thing - "the given permissions", not "a given permission".

File: kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/checks/MemberChecks.kt [hunk 3]
```diff
@@ -113,3 +115,100 @@ public suspend fun CheckContext<*>.notHasPermission(perm: Permission) {
...
+            logger.passed()
+
+            pass()
+        } else {
+            logger.failed("Member $member does not have permissions $perms")
+
+            fail(
+                translate(
+                    "checks.hasPermissions.failed",
+                    replacements = arrayOf(perms.values.forEach { it.translate(locale) })
```
Comment (gdude2002) [inline kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/checks/MemberChecks.kt]: This needs to be joined with a comma and space, it's not going to translate properly like this.

File: kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/checks/MemberChecks.kt [hunk 4]
```diff
@@ -113,3 +115,100 @@ public suspend fun CheckContext<*>.notHasPermission(perm: Permission) {
...
+            else -> memberObj.hasPermissions(perms.values)
+        }
+
+        if (result) {
+            logger.failed("Member $member has permissions $perms")
+
+            fail(
+                translate(
+                    "checks.notHasPermissions.failed",
+                    replacements = arrayOf(perms.values.forEach { it.translate(locale) })
```
Comment (gdude2002) [inline kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/checks/MemberChecks.kt]: This needs to be joined with a comma and space, it's not going to translate properly like this.

**Review by gdude2002 — CHANGES_REQUESTED** [review]
Just a few small things.

**Review by gdude2002 — APPROVED** [review]

**Comment by gdude2002** [comment]
I'm not sure why you have the Minecraft facet in this project, but otherwise looks good. Thanks.

**Comment by NoComment1105** [comment]
Lmao, I think it's just added by the plugin for some reason, not sure either

**Pull request merged**